vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/NetworkActions.java

```diff
@@ -0,0 +1,184 @@
+import org.vividus.steps.ui.validation.IBaseValidations;
+import org.vividus.ui.action.search.Locator;
+import org.vividus.ui.mobile.action.search.AppiumLocatorType;
+
+import io.appium.java_client.android.AndroidDriver;
+import io.appium.java_client.android.connection.ConnectionStateBuilder;
+
+public class NetworkActions
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworkActions.class);
+
+    private static final String IOS_PREFERENCES_BUNDLE_ID = "com.apple.Preferences";
+
+    private final IWebDriverProvider webDriverProvider;
+    private final GenericWebDriverManager genericWebDriverManager;
+    private final ApplicationActions applicationActions;
+    private final IBaseValidations baseValidations;
+
+    public NetworkActions(IWebDriverProvider webDriverProvider, GenericWebDriverManager genericWebDriverManager,
+            ApplicationActions applicationActions, IBaseValidations baseValidations)
+    {
+        this.webDriverProvider = webDriverProvider;
+        this.genericWebDriverManager = genericWebDriverManager;
+        this.applicationActions = applicationActions;
+        this.baseValidations = baseValidations;
+    }
+
+    public void changeNetworkConnectionState(NetworkToggle networkToggle, Mode mode)
+    {
+        if (genericWebDriverManager.isAndroid())
+        {
+            webDriverProvider.getUnwrapped(AndroidDriver.class)
+                    .setConnection(NetworkToggle.ON == networkToggle ? mode.getEnablingConnectionStateBuilder().build()
+                            : mode.getDisablingConnectionStateBuilder().build());
+        }
+        if (genericWebDriverManager.isIOS())
+        {
+            changeNetworkConnectionStateForIOS(networkToggle, mode);
+        }
+    }
+
+    private void changeNetworkConnectionStateForIOS(NetworkToggle networkToggle, Mode mode)
+    {
+        applicationActions.activateApp(IOS_PREFERENCES_BUNDLE_ID);
+        findElement(new Locator(AppiumLocatorType.ACCESSIBILITY_ID, mode.getIOSNetworkConnectionAlias()))
+                .ifPresent(WebElement::click);
+        findElement(new Locator(AppiumLocatorType.IOS_CLASS_CHAIN,
+                String.format("**/XCUIElementTypeSwitch[`label == \"%s\"`]", mode.getIOSNetworkConnectionAlias())))
+                        .ifPresent(switchButton ->
+                        {
+                            if (!networkToggle.active(switchButton.getAttribute("value")))
+                            {
+                                switchButton.click();
+                            }
+                            else
+                            {
+                                LOGGER.atInfo().addArgument(mode).addArgument(networkToggle).log("{} is already {}.");
+                            }
+                        });
+    }
+
+    private Optional<WebElement> findElement(Locator locator)
+    {
+        return baseValidations.assertElementExists("The element to click", locator);
```

Comment:
It makes sense to be specific about what element we are searching for, in case of failure it would be not clear for the user what is element to click

Reply:
@VitaliaPiliuhina  still not fixed